docs/migration/cosmoshub-4-delta-upgrade.md

```diff
@@ -0,0 +1,124 @@
+
+### Testing
+
+For those validator and full node operators that are interested in ensuring preparedness for the impending upgrade, complete and detailed testing instructions are provided in the [gravity-dex-upgrade-test](https://github.com/b-harvest/gravity-dex-upgrade-test/) Github repository. This repository has been tested by members of the core Cosmos ecosystem, as well as ecosystem partners which include validators, exchanges, and service providers.
+
+### Public testnet
+
+Validator and full node operators that wish to test their systems on a public testnet are encouraged to join the Tendermint team's public testnet, described [here](https://github.com/b-harvest/gravity-dex-upgrade-test/#public-testnet-info).
+
+### Current runtime, cosmoshub-4 (pre-Delta upgrade) is running Gaia v4.2.1
+
+The Cosmos Hub mainnet network, `cosmoshub-4`, is currently running [Gaia v4.2.1](https://github.com/cosmos/gaia/releases/tag/v4.2.1). We anticipate that operators who are running earlier versions of Gaia, e.g., v4.2.x, will be able to upgrade successfully; however, this is untested and it is up to operators to ensure that their systems are capable of performing the upgrade. 
+
+### Target runtime, cosmoshub-4 (post-Delta upgrade) will run Gaia v5.0.0
+
+The Comsos Hub mainnet network, `cosmoshub-4`, will run [Gaia v5.0.0](https://github.com/cosmos/gaia/releases/tag/v5.0.0). Operators _MUST_ use this version post-upgrade to remain connected to the network. 
+
+## Delta upgrade steps
+
+The following steps assume that an operator is running v4.2.1 (running an earlier version is untested). The upgrade has only been tested with v4.2.1 and these instructions follow this prerequisite.
+
+1. Prior to the upgrade, operators _MUST_ be running Gaia v4.2.1.
+2. At the upgrade block height of [6910000](#Upgrade-will-take-place-July-12,-2021), the Gaia software will panic with a message similar to the below:
+
+> ERR UPGRADE "Gravity-DEX" NEEDED at height: 6910000: v4.2.1
+> 
+> ERR CONSENSUS FAILURE!!! err="UPGRADE \"Gravity-DEX\" NEEDED at height: 6910000: v4.2.1" module=consensus stack="goroutine ...
+
+3. Replace the Gaia v4.2.1 binary with the Gaia v5.0.0 binary
+4. Start the Gaia v5.0.0 binary using the following command (also applying any additional flags and parameters to the binary needed by the operator, e.g., `--home $HOME`):
+
+> gaiad start --x-crisis-skip-assert-invariants
+
+IMPORTANT: The flag `--x-crisis-skip-assert-invariants` is optional and can be used to reduce memory and processing requirements while the in-place ugprade takes place before resuming connecting to the network.
+
+5. Wait until 2/3+ of voting power has upgraded for the network to start producing blocks
+
+## Upgrade duration
+
+The upgrade may take several hours to take complete because cosmoshub-4 participants operate globally with differing operating hours and it may take some time for operators to upgrade their binaries and connect to the network.
```

Comment:
```suggestion
The upgrade may take several hours to complete because cosmoshub-4 participants operate globally with differing operating hours and it may take some time for operators to upgrade their binaries and connect to the network.
```